app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -45,6 +47,23 @@ def self.default_capsule
         end
       end
 
+      def puppet_path
+        self[:puppet_path] || update_puppet_path
+      end
+
+      def update_puppet_path
+        return unless has_feature?(PULP_NODE_FEATURE)
```

Comment:
I tested and it works on an external capsule, but not the Katello.  Katello's have `PULP_FEATURE` not the Node feature and would need to hit ProxyAPI::Pulp instead.